block/validate.go

```diff
@@ -32,10 +33,16 @@ func (m *Manager) SettlementValidateLoop(ctx context.Context) error {
 				// get next batch that needs to be validated from SL
 				batch, err := m.SLClient.GetBatchAtHeight(currH)
 				if err != nil {
-					// TODO: should be recoverable. set to unhealthy and continue
-					return err
+					m.logger.Error("Getting batch from SL", "height", currH, "error", err)
+					if m.IsHealthy() {
+						err := fmt.Errorf("Unable to get state from SL. Err:%w", err)
+						m.setUnhealthy(err)
+					}
+					continue
+				}
+				if !m.IsHealthy() {
+					m.setHealthy()
 				}
```

Comment:
not sure I understand the logic here..
by this logic, If I can't get the batch than I set node to unhealthy, and skip current validation height, only to try to get next height, which if is ok, than I continue as if nothing happend and set node to healthy again?

so my questions are:
1. what about the previous height? 
2. seems like the error is unrecoverable from the calling function (i.e `GetBatchAtHeight`) so why do we continue?
3. doesnt it make sense that the node actually freezes like it was before? we just need to make sure the operator understand (cause for example if he's using bad RPC not sure we want the node to continue syncing and skip validation - but rather change rpc
)